File: tools/bench/answers/go_haiku/s1/05_dedup.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	nums := []int{3, 1, 3, 2, 1, 4}
	seen := make(map[int]bool)
	var result []string
	for _, num := range nums {
		if !seen[num] {
			seen[num] = true
			result = append(result, strconv.Itoa(num))
		}
	}
	fmt.Println(strings.Join(result, " "))
}

File: tools/bench/answers/rust_sonnet/s2/05_dedup.rs
fn main() {
    let list = [3, 1, 3, 2, 1, 4];
    let mut seen = Vec::new();
    let mut result = Vec::new();
    for &x in &list {
        if !seen.contains(&x) {
            seen.push(x);
            result.push(x.to_string());
        }
    }
    println!("{}", result.join(" "));
}
